tests/index.js

```diff
@@ -34,6 +34,63 @@ describe('broccoli-simple-replace', function(){
     });
   })
 
+  it('can use a regular expression in files', function(){
+    var sourcePath = 'tests/fixtures/string-for-string';
+    var tree = replace(sourcePath, {
+      files: [ new RegExp('matched' + '(.*js)') ],
+      pattern: {
+        match: 'REPLACE_ME',
+        replacement: 'REPLACED!!!'
+      }
+    });
+
+    builder = new broccoli.Builder(tree);
+    return builder.build().then(function(dir) {
+      var actual = fs.readFileSync(dir + '/matched-file.js', { encoding: 'utf8'});
+      var expected = 'ZOMG, ZOMG\n\nREPLACED!!!\n\nZOMG, ZOMG\n';
+
+      expect(actual).to.equal(expected);
+    });
+  })
+
+  it('can use a glob in files', function(){
+    var sourcePath = 'tests/fixtures/string-for-string';
+    var tree = replace(sourcePath, {
+      files: [ '*.js' ],
+      pattern: {
+        match: 'REPLACE_ME',
+        replacement: 'REPLACED!!!'
+      }
+    });
+
+    builder = new broccoli.Builder(tree);
+    return builder.build().then(function(dir) {
+      var actual = fs.readFileSync(dir + '/matched-file.js', { encoding: 'utf8'});
+      var expected = 'ZOMG, ZOMG\n\nREPLACED!!!\n\nZOMG, ZOMG\n';
+
+      expect(actual).to.equal(expected);
+    });
+  })
+
+  it('can use a function in files', function(){
+    var sourcePath = 'tests/fixtures/string-for-string';
+    var tree = replace(sourcePath, {
+      files: [ function(x) { return 'matched.js'; }  ],
```

Comment:
The idea here is that the function gets the file are being processed, and returns true or false if it matches/doesn't match. Can you tweak this to return true, and add another test that always returns false (causing no files to be changed)?